src/main/java/com/crowdin/action/UploadTranslationsAction.java

```diff
@@ -1,71 +1,75 @@
 package com.crowdin.action;
 
 import com.crowdin.client.Crowdin;
+import com.crowdin.client.CrowdinProperties;
+import com.crowdin.client.CrowdinPropertiesLoader;
 import com.crowdin.client.languages.model.Language;
 import com.crowdin.client.sourcefiles.model.Branch;
 import com.crowdin.client.sourcefiles.model.Directory;
 import com.crowdin.client.sourcefiles.model.File;
 import com.crowdin.util.*;
-import com.intellij.notification.NotificationGroup;
-import com.intellij.openapi.actionSystem.AnAction;
 import com.intellij.openapi.actionSystem.AnActionEvent;
 import com.intellij.openapi.project.Project;
 import com.intellij.openapi.vfs.VirtualFile;
-import com.intellij.openapi.wm.StatusBarWidget;
-import com.intellij.openapi.wm.StatusBarWidgetProvider;
 import org.jetbrains.annotations.NotNull;
-import org.jetbrains.annotations.Nullable;
 
 import java.util.List;
 import java.util.Map;
-
-import static com.crowdin.util.PropertyUtil.PROPERTY_SOURCES;
+import java.util.concurrent.atomic.AtomicInteger;
 
 public class UploadTranslationsAction extends BackgroundAction {
 
-    private static final String PATTERN = "/values-%android_code%/%original_file_name%";
-
     @Override
     public void performInBackground(@NotNull AnActionEvent e) {
         Project project = e.getProject();
         VirtualFile root = project.getBaseDir();
-        String sourcesProp = PropertyUtil.getPropertyValue(PROPERTY_SOURCES, project);
-        List<String> sourcesList = FileUtil.getSourcesList(sourcesProp);
-        Crowdin crowdin = new Crowdin(project);
 
-        List<Language> projectLanguages = crowdin.getProjectLanguages();
+        CrowdinProperties properties;
+        try {
+            properties = CrowdinPropertiesLoader.load(project);
+            Crowdin crowdin = new Crowdin(project, properties.getProjectId(), properties.getApiToken(), properties.getBaseUrl());
+
+            List<Language> projectLanguages = crowdin.getProjectLanguages();
 
-        String branch = GitUtil.getCurrentBranch(project);
-        Long branchId = crowdin.getBranch(branch).map(Branch::getId).orElse(null);
+            String branch = properties.isDisabledBranches() ? "" : GitUtil.getCurrentBranch(project);
+            Long branchId = crowdin.getBranch(branch).map(Branch::getId).orElse(null);
 
-        List<com.crowdin.client.sourcefiles.model.File> files = crowdin.getFiles(branchId);
-        Map<Long, Directory> dirs = crowdin.getDirectories(branchId);
-        Map<String, File> filePaths = CrowdinFileUtil.buildFilePaths(files, dirs);
+            List<com.crowdin.client.sourcefiles.model.File> files = crowdin.getFiles(branchId);
+            Map<Long, Directory> dirs = crowdin.getDirectories(branchId);
+            Map<String, File> filePaths = CrowdinFileUtil.buildFilePaths(files, dirs);
 
-        int uploadedFilesCounter = 0;
+            AtomicInteger uploadedFilesCounter = new AtomicInteger(0);
```

Comment:
Why we need `AtomicInteger`. Are we introducing any concurrency here? We already in separate thread because we in `performInBackground` but here we do not do anything in multiple thread. Or am I wrong?

Reply:
It is used for anonymous functions. Anonymous functions work only with final external values.